### ISSUE TITLE
Auth_token 

### ISSUE THREAD

**Issue by Imd5** [Jamonek/Robinhood#117]
Could some one explaine to me how to save authorization token.

**Comment by jasonleehodges** [comment]
See discussion in #71. Are you looking to store it locally? If so, are you looking to save it as an environment variable? A hashed value in a database? Or, more dangerously, in plain text?

**Comment by Imd5** [comment]
I did but didn’t see ). I want to store it in plain text

**Comment by jasonleehodges** [comment]
Once you've logged in via one of the authentication methods (like `login()` or `login_prompt()`) then the auth_token property of your class is set by default. You can take that value and write it to a plain text file or a csv or anything you want really. For example:

```
from Robinhood import Robinhood

rb = Robinhood()
rb.login({username},{password})

file = open(“testfile.txt”, ”w”) 
file.write(rb.auth_token)
file.close()
```

Of course, storing tokens in plain text is not advisable because if someone stumbles across this file they essentially have access to your username and password and any privileges that come with that.

**Comment by Imd5** [comment]
Thank you. RH has limit on number authorization per second. So I’m trying to bypass it)

**Comment by jasonleehodges** [comment]
Yeah, you shouldn't have to login every time you make a query to one of their endpoints. The initial login opens a session that will stay open and authenticated as long as your script is running. So you can run a loop that calls whatever endpoint you are interested in after the authentication, just make sure you don't re-authenticate within the loop and then you don't have to store the token. For example:

```
from Robinhood import Robinhood

rb = Robinhood()
rb.login({username},{password})

for i in range(0,1000):
    rb.portfolios()
```

That will get all your portfolio data 1000 times but only authenticate once.

**Comment by Imd5** [comment]
It isn’t about my portfolio. I try to check all tradable stock  on RH. Then next step to implement some strategy.

**Comment by jasonleehodges** [comment]
Well you don't even need to authenticate just to check a quote.
